crates/sui-types/src/temporary_store.rs

```diff
@@ -351,12 +570,71 @@ impl<S> TemporaryStore<S> {
         // The adapter is not very disciplined at filling in the correct
         // previous transaction digest, so we ensure it is correct here.
         object.previous_transaction = self.tx_digest;
-        self._written.insert(object.id(), (object, kind));
+        self.written
+            .insert(object.id(), (ctx.clone(), object, kind));
     }
 
-    pub fn delete_object(&mut self, id: &ObjectID, version: SequenceNumber, kind: DeleteKind) {
+    pub fn charge_gas(
```

Comment:
in gas.rs instead?

Reply:
moving this to gas.rs will be a big change, there are lot of dependency of temp_store and might need a lot of changes to do this cleanly, I suggest doing this in a different PR